core/java/iesi-core/src/main/java/io/metadew/iesi/component/http/HttpComponentDefinitionService.java

```diff
@@ -21,15 +24,34 @@ public synchronized static HttpComponentDefinitionService getInstance() {
         return INSTANCE;
     }
 
-    private HttpComponentDefinitionService() {
+    private HttpComponentDefinitionService() { }
+
+    public HttpComponentHeaderDesign convertHeaders(HttpHeaderDefinition httpHeaderDefinition, UUID id) {
+        UUID uuid = UUID.randomUUID();
+        return new HttpComponentHeaderDesign(
+                uuid,
+                new HttpComponentDesignTraceKey(id),
+                httpHeaderDefinition.getName(),
+                httpHeaderDefinition.getValue()
+        );
+    }
+
+    public HttpComponentQueryDesign convertQueries(HttpQueryParameterDefinition httpQueryParameterDefinition, UUID id) {
+        UUID uuid = UUID.randomUUID();
+        return new HttpComponentQueryDesign(
+                uuid,
+                new HttpComponentQueryDesignKey(id),
+                httpQueryParameterDefinition.getName(),
+                httpQueryParameterDefinition.getValue()
+        );
```

Comment:
could this be exported to a dedicated class 'HttpQueryParameterDesignTraceService' with method 'public HttpQueryParameterDesignTrace convert(HttpQueryParameterDefinition httpQueryParameterDefinition, ActionExecution actionExecution, String actionParameterName);

could this be exported to a dedicated class 'HttpHeaderDesignTraceService' with method 'public HttpHeaderDesignTrace convert(HttpHeaderDefinition httpHeaderDefinition, ActionExecution actionExecution, String actionParameterName);